happi/cli.py

```diff
@@ -252,6 +257,19 @@ def happi_cli(args):
 
         from IPython import start_ipython  # noqa
         start_ipython(argv=['--quick'], user_ns=devices)
+    elif args.cmd == "update":
+        # parse input
+        input_ = " ".join(args.json).strip()
+        if input_ == "-":
+            items_input = json.load(sys.stdin)
+        else:
+            items_input = json.loads(input_)
+        # insert
+        registry = happi.containers.registry
+        for item in items_input:
+            item = client.create_device(device_cls=item["type"], **item)
+            exists = item["_id"] in [c["_id"] for c in client.all_items]
+            client._store(item, insert=not exists)
```

Comment:
happi has some weird saving methodology, where it patches on `save` to the ~device~ item:
https://pcdshub.github.io/happi/master/client.html#editing-device-information

I think the original API intends for this to be the only call you need:
```suggestion
            item.save()
```

I suppose this may fail with the scenario `insert=True/exists=True` and as such the proposed `exists_ok` kwarg?

Reply:
Correct. This was the least invasive solution I could find to get my desired behavior of overwrite if exists. Your suggestion does work for new items.